client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -126,11 +129,32 @@ tie.factory('PythonCodePreprocessorService', [
         while (true) {
           var codeToAdd = '';
           lastFunctionNameLocation = code.indexOf(
-            functionName, lastFunctionNameLocation);
+            functionNameWithoutParen, lastFunctionNameLocation);
           if (lastFunctionNameLocation === -1) {
             break;
           }
           if (lastFunctionNameLocation !== functionNameStart) {
+            var whitespaceCheckLocation = (
+              lastFunctionNameLocation + functionNameWithoutParen.length);
+            // We assume we found the matched function, but we might have found
+            // a similarly-named other function / variable. User-inputted code
+            // can be ... tricky.
+            var isMatchedFunction = true;
+            // We can safely assume there's a left paren because we matched it
+            // with the regular expression.
+            while (code[whitespaceCheckLocation] != '(') {
+              var isValidWhitespace = VALID_WHITESPACE_REGEX.test(
```

Comment:
Ah, OK, I see, thanks! You might want to add this sort of thing to the Karma test, it feels like a breakage can easily occur here.

I think the reasoning is sound, but would suggest adding a comment to the code explaining this case in particular ("X can be ... tricky" doesn't quite communicate the nuances). Also maybe create a separate isWhitespaceChar(c) method.

Also, do you need a check for "end of student code" in the "while" clause so that there's a guarantee of termination?